libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -431,10 +433,22 @@ - (void)continueNativeBrowserFlow {
     [SFSDKCoreLogger d:[self class] format:@"%@: Initiating native browser flow with URL %@", NSStringFromSelector(_cmd), approvalUrl];
     NSURL *nativeBrowserUrl = [NSURL URLWithString:approvalUrl];
     [SFSDKAppFeatureMarkers registerAppFeature:kSFAppFeatureSafariBrowserForLogin];
-    SFSafariViewController *svc = [[SFSafariViewController alloc] initWithURL:nativeBrowserUrl];
-    svc.delegate = self;
-    self.advancedAuthState = SFOAuthAdvancedAuthStateBrowserRequestInitiated;
-    [self.delegate oauthCoordinator:self didBeginAuthenticationWithSafariViewController:svc];
+    __weak typeof(self) weakSelf = self;
+    _authSession = [[SFAuthenticationSession alloc] initWithURL:nativeBrowserUrl callbackURLScheme:nil completionHandler:^(NSURL * _Nullable callbackURL, NSError * _Nullable error) {
+        if (error) {
+            [weakSelf.delegate oauthCoordinatorDidCancelBrowserAuthentication:self];
+        }
+        else {
+            [[SFUserAccountManager sharedInstance] handleAdvancedAuthenticationResponse:callbackURL options:nil];
```

Comment:
@wmathurin Can we do away with the dependency on SFUserAccountManager? Is it possible to perhaps invoke a completion block instead or  we could fire a notification?

Reply:
SFUserAccountManager's handleAdvancedAuthenticationResponse calls SFSDKURLHandlerManager's processRequest which calls SFSDKAdvancedAuthURLHandler's processRequest which calls SFUserAccountManager's handleNativeAuthResponse which goes to SFSDKOAuthClient's handleURLAuthenticationResponse which goes back to SFOAuthCoordinator's handleAdvancedAuthenticationResponse !

So I guess I could call the SFOAuthCoordinator method directly.

Reply:
Done [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2621/commits/343477a4e708bd7b298140c131bba7542b04edf8)